status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,409 @@
+
+Application level functions are defined in the _application_ layer.
+Status currently defines functionality to support three main application features:
+
+- Status Communities, as specified in [56/STATUS-COMMUNITIES](../56/communities.md)
+- Status 1:1 Chat, as specified in [55/STATUS-1TO1-CHAT](../55/1to1-chat.md)
+- Status Private Group Chat, as specified in a subsection of [55/STATUS-1TO1-CHAT](../55/1to1-chat.md#negotiation-of-a-11-chat-amongst-multiple-participants-group-chat)
+
+<!-- TODO: list functions not related to main app features, such as user sync, backup, push notifications, etc. -->
+
+Each application-level function, regardless which feature set it supports, has the following properties:
+
+1. Functional scope
+1. Content topic
+1. Ephemerality
+
+### Functional Scope
+
+Each Status app-level message MUST define a functional scope.
+The functional scope MUST define the _minimum_ scope of the audience that should _participate_ in the app function the message is related to.
+In other words, it determines the minimum subset of Status app participants
+that should have access to messages related to that function.
+
+Note that the functional scope is distinct from the number of participants that is _addressed_ by a specific message.
+For example, a participant will address a 1:1 chat to only one other participant.
+However, since all users of the Status app MUST be able to participate in 1:1 chats,
+the functional scope of messages enabling 1:1 chats MUST be a global scope.
+Similarly, since private group chats can be set up between any subset of Status app users,
+the functional scope for messages related to private group chats MUST be global.
+As a counter-example, messages that originate within a community (and are _addressed_ to members of that community)
+are only of interest to participants that are also members of that community.
+Such messages MUST have a community-wide functional scope.
+A third group of messages are addressed only to the participant that generated those messages itself.
+These _self-addressed_ messages MUST have a local functional scope.
+
+If we further make a distinction between "control" and "content" messages,
+we can distinguish five distinct functional scopes.
+
+All Status messages MUST have one of these functional scopes:
+
+#### Global scope
+
+1. _Global control_: messages enabling the basic functioning of the app to control features that all app users should be able to participate in. Examples include Contact Requests, Community Invites, global Status Updates, Group Chat Invites, etc.
+2. _Global content_: messages carrying user-generated content for global functions. Examples include 1:1 chat messages, images shared over private group chats, etc.
+
+#### Community scope
+
+3. _Community control_: messages enabling the basic functioning of the app to control features _only relevant to members of a specific community_. Examples include Community Membership Updates, community Status Updates, etc.
```

Comment:
Thanks! I imagine this is based on some community description messages that are published "to everyone" - i.e. at least on shards that everyone is subscribed to and is not encrypted by the community key? In this case, these would still be modelled as "global" messages as they affect operations that everyone should be able to participate in. "Community control" refers to those messages that _can_ be distributed only within a single community.

Reply:
Yes, that information is derived from `CommunityDescription`. Please note that `CommunityDescription` itself is never fully encrypted; only specific sub-elements, like members and channels, are encrypted depending on the context. Even so, these cards will render correctly, with the exception of member counts and active member counts, which remain inaccessible to users without the necessary key.

As for shards, `CommunityDescription` can reside within its own community shard. The protocol is designed to ensure that everyone can locate the shard a given community uses, enabling them to subscribe to that shard to access the description. Essentially, the community owner publishes `CommunityDescription` on the community shard and shares shard details on the default shard.
Rationale: https://github.com/status-im/status-go/issues/3961#issuecomment-1785453710
Issue: https://github.com/status-im/status-go/issues/4230
Implementation: https://github.com/status-im/status-go/pull/4499

Reply:
Thanks! I think the issue here might be that _eventually_ we'd want Communities to be able to provide their own infrastructure - their own bootstrap nodes, Store nodes, etc. In this case we may not _want_ everyone to be able to retrieve the Community Description (which would require discovering and using that Communities internal service infrastructure). Of course, it will be easy to define a global shard only used for "discoverable" Community descriptions, so I don't think the spec prohibits either design.